akka-actor-tests/src/test/scala/akka/pattern/BackoffOnRestartSupervisorSpec.scala

```diff
@@ -161,7 +161,7 @@ class BackoffOnRestartSupervisorSpec extends AkkaSpec with ImplicitSender {
 
       supervisor ! BackoffSupervisor.GetCurrentChild
       // new instance
-      val Some(child) = expectMsgType[BackoffSupervisor.CurrentChild].ref
```

Comment:
Is this style of pattern patching deprecated now?

Reply:
Yes, it's only allowed for exhaustive matches now (I'm pretty sure that's intentional, though I can't find a reference right now).

Which makes sense for production code, but for tests it was kinda nice ;)

Reply:
oh, @dwijnand found `-Xlint:-valpattern,_`, which we could enable for the tests

Reply:
I'd `.get` instead of globally disabling, but maybe global disable first and `.get` incrementally.